packages/flutter/lib/src/cupertino/segmented_control.dart

```diff
@@ -0,0 +1,683 @@
+    _children.forEach(visitor);
+  }
+
+  static BoxParentData _boxParentData(RenderBox box) => box.parentData;
+
+  static Size _boxSize(RenderBox box) => box == null ? Size.zero : box.size;
+
+  static Rect _boxRect(RenderBox box) {
+    return box == null ? Rect.zero : _boxParentData(box).offset & box.size;
+  }
+
+  static double _minWidth(RenderBox box, double height) {
+    return box == null ? 0.0 : box.getMinIntrinsicWidth(height);
+  }
+
+  static double _minHeight(RenderBox box, double width) {
+    return box == null ? 0.0 : box.getMinIntrinsicHeight(width);
+  }
+
+  @override
+  double computeMinIntrinsicWidth(double height) {
+    double total = 0.0;
+    for (RenderBox box in boxes) {
+      total += _minWidth(box, height);
+    }
+    return total;
+  }
+
+  @override
+  double computeMaxIntrinsicWidth(double height) => computeMinIntrinsicWidth(height);
+
+  @override
+  double computeMinIntrinsicHeight(double width) => _minHeight(boxes.elementAt(0), width);
+
+  @override
+  double computeMaxIntrinsicHeight(double width) => computeMinIntrinsicHeight(width);
+
+  @override
+  double computeDistanceToActualBaseline(TextBaseline baseline) {
+    return boxes.elementAt(0).getDistanceToActualBaseline(baseline);
+  }
+
+  @override
+  void performLayout() {
+    double maxHeight = _kMinSegmentedControlHeight;
+    final Constraints parentConstraints = constraints.loosen();
+
+    for (RenderBox box in boxes) {
+      box.layout(parentConstraints, parentUsesSize: true);
+      final double boxHeight = box.size.height;
+      if (boxHeight > maxHeight) {
+        maxHeight = boxHeight;
+      }
+    }
+
+    final double overallWidth = constraints.maxWidth;
+    final Size overallSize =
+        constraints.constrain(Size(overallWidth, maxHeight));
+
+    Offset centerLayout(Size boxSize, double x) {
+      assert(maxHeight >= boxSize.height);
+      final Offset boxOffset =
+          new Offset(x, (maxHeight - boxSize.height) / 2.0);
+      return boxOffset;
+    }
+
+    final double childWidth = overallSize.width / boxes.length;
+
+    final BoxConstraints childConstraints =
+        new BoxConstraints.tightFor(width: childWidth, height: maxHeight);
+
+    for (RenderBox box in boxes) {
+      box.layout(childConstraints, parentUsesSize: true);
+    }
+
+    Offset childOffset = Offset.zero;
+
+    const double left = 0.0;
+    final double right = overallSize.width;
+
+    switch (textDirection) {
+      case TextDirection.rtl:
+        double start = right;
+        for (int i = 0; i < boxes.length; i += 1) {
+          childOffset = centerLayout(Size(childWidth, maxHeight), start-childWidth);
+          _boxParentData(boxes.elementAt(i)).offset = childOffset;
+
+          final Rect childRect =
+              Rect.fromLTWH(start-childWidth, 0.0, childWidth, maxHeight);
+          RRect rChildRect;
+          if (i == 0) {
+            rChildRect = RRect.fromRectAndCorners(childRect,
```

Comment:
throughout this code, please avoid any abbreviations (`i`, `rChildRect`, etc). It should be possible for someone to drop into the middle of the code and figure out what all the variables mean without having to browse through the code to figure out what each variable expands to.